### PR TITLE
Fix #1970: Fix links to JavaDoc

### LINKED ISSUE #1970: Fix links to JavaDoc
Links to the JavaDoc are broken.

See http://stackoverflow.com/questions/16934488/how-to-link-classes-from-jdk-into-scaladoc-generated-doc/ for a fix.

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
Review by @sjrd

**Comment by sjrd** [comment]
Could the CSS fille be somewhere else than at the top level of the repo?

File: project/Build.scala [hunk 1]
```diff
@@ -135,9 +137,69 @@ object Build extends sbt.Build {
             .find(_.endsWith(java.io.File.separator + "rt.jar")).get
         }
 
-        file(rtJar) -> url("http://docs.oracle.com/javase/8/docs/api/")
-      }
+        file(rtJar) -> url(javaDocBaseURL)
+      },
+
+
```
Comment (sjrd) [inline project/Build.scala]: Double blank line

**Comment by gzm0** [comment]
> Could the CSS fille be somewhere else than at the top level of the repo? 

Yes, of course :) Anywhere you like. I just haven't found any better place (short of making a new directory for a single file).

**Comment by sjrd** [comment]
> short of making a new directory for a single file

Better that than showing it on the top level.

File: project/Build.scala [hunk 2]
```diff
@@ -135,9 +137,69 @@ object Build extends sbt.Build {
...
+      /* Patch the ScalaDoc we generate.
+       *
+       *  After executing the normal doc command, copy everything to the
+       *  `patched-api` directory (same internal directory structure) while
+       *  patching the following:
+       *
+       *  - Append `additional-doc-styles.css` to `lib/template.css`
+       *  - Fix external links to the JavaDoc. I.e. change
```
Comment (sjrd) [inline project/Build.scala]: JavaDoc**, i.e.,** change

File: project/Build.scala [hunk 3]
```diff
@@ -135,9 +137,69 @@ object Build extends sbt.Build {
...
+        // Where to store the patched docs
+        val outDir = crossTarget.value / "patched-api"
+
+        // Find all files in the current docs
+        val docPaths = {
+          val docDir = (doc in Compile).value
+          Path.selectSubpaths(docDir, new SimpleFileFilter(_.isFile)).toMap
+        }
+
+        // File with our CSS styles (need canonical so comparison below works)
```
Comment (sjrd) [inline project/Build.scala]: so **that the** comparison works?

File: project/Build.scala [hunk 4]
```diff
@@ -135,9 +137,69 @@ object Build extends sbt.Build {
...
+      /* Patch the ScalaDoc we generate.
+       *
+       *  After executing the normal doc command, copy everything to the
+       *  `patched-api` directory (same internal directory structure) while
+       *  patching the following:
+       *
+       *  - Append `additional-doc-styles.css` to `lib/template.css`
+       *  - Fix external links to the JavaDoc. I.e. change
+       *    `${javaDocBaseURL}index.html#java.lang.String` to
+       *    `${javaDocBaseURL}index.html?java/lang/String.html`
```
Comment (sjrd) [inline project/Build.scala]: What if we link to a *member* of a class? Or to a "static" method, which would be in a companion object in Scala but as static in Java?

**Comment by sjrd** [comment]
That's all.

File: project/Build.scala (hunk 4 above)
Comment (gzm0) [inline project/Build.scala]: ScalaDoc doesn't let us do that from what I have tried...

Comment (sjrd) [inline project/Build.scala]: If you say so ... ^^ OK

**Comment by gzm0** [comment]
Any idea where we could put that css? I'm not very inspired...

**Comment by sjrd** [comment]
`css/`? `assets/`? `doc-hacks/`? `scaladoc-assets/`?

**Comment by gzm0** [comment]
I guess `assets` is most reasonable :)

File: project/Build.scala (hunk 4 above)
Comment (gzm0) [inline project/Build.scala]: Seems that links to non-static members work... figure...

**Comment by gzm0** [comment]
Updated. Failure case is tested manually

**Comment by sjrd** [comment]
LGTM

**Pull request merged**